### PR TITLE
PS-8102: Fixing memory leak in keyring_common

### PR DESCRIPTION
**Pull request by dutow**
get_default_mr in keyring_common created a new instance of the default
memory resource every time the funciton was called instead of using a
single static instance.

This commit fixes this by making it a static variable.

### CODE REVIEW

**Review by altmannmarcelo — APPROVED** [review]
LGTM. Will cherry-pick to PXB

**Review by ldonoso — APPROVED** [review]

**Review by altmannmarcelo — APPROVED** [review]
LGTM

File: components/keyrings/common/data/global_default_mr.h [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 inline ::boost::container::pmr::memory_resource
     * ::boost::container::pmr::get_default_resource() BOOST_NOEXCEPT {
-  psi_memory_resource *global_default_mr = new psi_memory_resource{};
+  static psi_memory_resource *global_default_mr = new psi_memory_resource{};
```
Comment (ldonoso) [inline components/keyrings/common/data/global_default_mr.h]: Please, why not the following? On the other hand, is there not an already available `ps_memory_resource` published by the FW?

```
static psi_memory_resource global_default_mr;

inline ::boost::container::pmr::memory_resource
    * ::boost::container::pmr::get_default_resource() BOOST_NOEXCEPT {
  return &global_default_mr;
}
```

**Review by ldonoso — CHANGES_REQUESTED** [review]

File: components/keyrings/common/data/global_default_mr.h (hunk 1 above)
Comment (dutow) [reply components/keyrings/common/data/global_default_mr.h]: Please check PS-8004, as this leak is caused by the fix for that issue (the original code, before the fix was exactly what you suggested here].

**Review by ldonoso — APPROVED** [review]

**Review by ldonoso — CHANGES_REQUESTED** [review]
Please, do we have concurrency problems over `global_default_mr`?

**Comment by dutow** [comment]
https://ps80.cd.percona.com/job/percona-server-8.0-param/1189/

**Review by ldonoso — APPROVED** [review]

**Pull request merged**